parse.c

```diff
@@ -353,7 +354,12 @@ static float sample_id_to_time(const HEADER_st *header, uint32_t sample)
 	if (!header->channels_count)
 		return -1.0;
 
-	return header->channels[0]->timediv * 10.0 * sample / header->channels[0]->samples_count;
+	// return header->channels[0]->timediv * 10.0 * sample / header->channels[0]->samples_count;
+
+	// T.S. 05NOV2021    SDS-7102:        20 DIVs    span over      all samples in the file:
+	return header->channels[0]->timediv * 20.0 * sample / header->channels[0]->samples_file;
```

Comment:
Seems like you are doing code versioning in the file. History is what git is for, no need to keep the old code.

That being said, mind explaining the difference in behavior introduced by this change? What was wrong with the previous behavior? Was this working for one mode but not another? We need more context here :)

Reply:
I just sent you the code as it works for me.

Here I attach a Zip-Archive with example data to explain what I got with the original code and after the changes.
[Example.zip](https://github.com/bjonnh/owon-sds7102-protocol/files/7489281/Example.zip)
  500A_ON.bin                  SDS7102 wave data file
  500A_ON_Image.png     SDS7102 image screen-shot
  500A_ON_old.csv           original owon-parse output.csv
  500A_ON_new.csv          new owon-parse output.csv
  Old-ty-scaling.png            plot with old.csv data
  New-ty-scaling.png           plot with new csv.data
`
The screen-shot shows the signal starting at Ch1 baseline, so voltage = 0
At trigger it drops down by approx 5.5 divs to approx -55mV
The 3 glitches are approx 2.7 divs = 54 ms apart.

With the original scaling, the signal goes from +3 to -2.5 units and the glitches are 0.035 units apart.

The plot with the new csv file with t in s and U in V matches the screen-shot image.